File: CP website/WorkatTech/Arrays/Merge Sorted Subarrays.java
//https://workat.tech/problem-solving/practice/merge-sorted-subarrays
class Solution {
	void merge (int[] arr, int endIndex) {
		// add your logic here
		int i=0;
		int j=endIndex+1;
		int [] c = new int[arr.length];
		int k=0;
		
		while(i<=endIndex && j<arr.length){
			if(arr[i]<=arr[j]){
				c[k] = arr[i];
				i++;
			}
			else{
				c[k] = arr[j];
				j++;
			}
			k++;
		}
		
		while(i<=endIndex){
			c[k]=arr[i];
			i++;k++;
		}
		while(j<arr.length){
			c[k]=arr[j];
			k++;j++;
		}
		
		for(i=0;i<arr.length;i++){
			arr[i] = c[i];
		}
		
	}
}

File: CP website/WorkatTech/Arrays/MergeSortedSubarrays.py
# https://workat.tech/problem-solving/practice/merge-sorted-subarrays/
class Solution:
	def merge(self, arr: List[int], endIndex: int) -> List[int]:
		# add your logic here
		n = len(arr)
		s = [0]*n
		i = 0
		j = endIndex+1
		k = 0
		while(i<=endIndex and j<n):
			if(arr[i]<=arr[j]):
				s[k] = arr[i]
				i+=1
			else:
				s[k] = arr[j]
				j+=1
			k+=1
		while(i<=endIndex):
			s[k]=arr[i]
			i+=1
			k+=1
		while(j<n):
			s[k]=arr[j]
			j+=1
			k+=1
		return s

